lib/spring/configuration.rb

```diff
@@ -52,6 +53,10 @@ def project_root_path
       @project_root_path ||= find_project_root(Pathname.new(File.expand_path(Dir.pwd)))
     end
 
+    def quiet
+      @quiet ||= ENV.key?('SPRING_QUIET') 
```

Comment:
>It short-circuits in the config reader method via ||= instead of ||, allowing it to be set back to false even after spring starts, if this is necessary.

Correct me if I'm wrong, but at first glance it looks like that it won't be possible to set it back to `false` after setting it to `true` once, like:
```ruby
class DummySpring
  class << self
    attr_writer :quiet

    def quiet
      @quiet ||= ENV.key?('SPRING_QUIET') 
    end
  end
end

ENV["SPRING_QUIET"] = "lol"
DummySpring.quiet # => true
DummySpring.quiet = false
DummySpring.quiet # => still true because `ENV.key?('SPRING_QUIET') ` gets checked again as long as `@quiet` is falsey
```

Reply:
@nvasilevski  That's a good point. You could set it back to false via the `attr_writer`, but not via the `ENV` var.

But I guess I assumed that you wouldn't try to use the ENV var to change it while an app is executing - if you want to do that, just use the `attr_writer`.

In other words, the `ENV` var is only needed when you *_don't_* have access to modify the code or config at all, but still want to turn quiet mode on,

This is also explained by the new documentation:

> You can also set the initial state of the `quiet` configuration option to true
by setting the `SPRING_QUIET` environment variable before executing Spring.

Note it says "initial state" and "...before executing Spring".

So, I guess I think this is OK as is, especially given the documentation.

WDYT?

Reply:
>You could set it back to false via the attr_writer, but not via the ENV var.

In my example I'm showing that it's not possible to reset it using `writer` after setting it once to `true` because of how `||=` works. If we want to support that, reader will need to be implemented as:
```ruby
def quiet
    return @quiet if defined(@quiet)

    @quiet = ENV.key?('SPRING_QUIET')
end
```

Reply:
> In my example I'm showing that it's not possible to reset it using writer after setting it once to true because of how ||= works.

Ah, gotcha, you're right.

So ideally we can come up with a case where it can be changed from true to false or vice versa, via either the attr_writer or the ENV var.

I'll try to come up with a change that does that and then ping you to review it.

Thanks for your help and attention!

Reply:
Sorry, I got busy/lazy and never returned to finish up this thread.

But I see this got merged now, which is great! 

This thread describes a minor bug which should be fixed eventually, so I'll keep it on my TODO list - near the bottom, which realistically means "probably never", but hope springs eternal ;)